### PR TITLE
feat: add WnTooltip component

### LINKED ISSUE #142: Tooltip
https://www.figma.com/design/J9pCZpUhcm0MRs7dFX7LTN/01.-Application-Components?node-id=550-69&m=dev

### PR DESCRIPTION
**Pull request by erskingardner**
## Summary
- Add `WnTooltip` widget with 4 positions (top, bottom, left, right)
- Support hover and long-press to show tooltip
- Tap outside to dismiss
- Configurable arrow visibility and wait duration
- Uses hooks for state management (no StatefulWidget)
- Add comprehensive test suite with 99% coverage (16 tests)
- Add widgetbook story with interactive playground and variant examples

Closes #142

## Test Plan
- [x] All 1463 Flutter tests pass
- [x] Precommit checks pass
- [x] Manual testing completed

## Summary by CodeRabbit

* **New Features**
  * New WnTooltip: configurable tooltip with top/bottom/left/right placement, tap/long-press/hover triggers, optional arrow, show delay, custom content, backdrop dismissal and fade animation; replaces prior tooltip usage in the UI.

* **Documentation**
  * Interactive showcase demonstrating positions and content variants.

* **Tests**
  * Comprehensive test suite covering rendering, triggers, positioning, arrow behavior, content precedence, dismissal, and edge cases.

* **Chores**
  * Raised precommit test coverage requirement to 95%.

### CODE REVIEW

File: lib/widgets/wn_tooltip.dart [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+
+    path.close();
+    canvas.drawPath(path, paint);
+  }
+
+  @override
+  bool shouldRepaint(covariant ArrowPainter oldDelegate) {
+    return oldDelegate.position != position || oldDelegate.color != color;
+  }
+}
```
[the remark this answers is not in the document]
Comment (erskingardner) [reply lib/widgets/wn_tooltip.dart]: I don't agree. since tooltips are positional and contextual from where they are triggered I believe it would make it more complex to manage it as a hook. you'd end up having to do a bunch of the same work over and over again in each calling location.
